src/external-types/reveal.js/index.d.ts

```diff
@@ -4,9 +4,199 @@
+
+  export type ControlsLayout = 'bottom-right' | 'edges'
+
+  export type NavigationMode = 'default' | 'linear' | 'grid'
+
+  export type ShowSlideNumber = 'all' | 'print' | 'speaker'
+
+  export type SlideNumber = 'h.v' | 'h/v' | 'c' | 'c/t'
+
+  export type TransitionSpeed = 'default' | 'slow' | 'fast'
+
   export interface RevealOptions {
-    plugins?: string[]
+    autoAnimate?: boolean,
+    // Time between auto-animate transitions
+    autoAnimateDuration?: number,
+    // Auto-animate transition function
+    autoAnimateEasing?: 'ease', // ToDo: find type for https://developer.mozilla.org/en-US/docs/Web/CSS/easing-function
+    /* autoAnimateMatcher?: null, // ToDo: I think this is not needed */
+    // CSS properties that can be auto-animated. Position & scale
+    // is matched separately so there's no need to include styles
+    // like top/right/bottom/left, width/height or margin.
+    autoAnimateStyles?: AutoAnimateStyles[],
+    // Should unmatched elements get auto-animate transitions
+    autoAnimateUnmatched?: boolean, // ToDo: If we omit autoAnimateMatcher, we don't need this do we?
+    // Global override for autolaying embedded media (video/audio/iframe)
+    // - null:   Media will only autoplay if data-autoplay is present
+    // - true:   All media will autoplay, regardless of individual setting
+    // - false:  No media will autoplay, regardless of individual setting
+    autoPlayMedia?: boolean | null,
+    // Controls automatic progression to the next slide
+    // - 0:      Auto-sliding only happens if the data-autoslide HTML attribute
+    //           is present on the current slide or fragment
+    // - 1+:     All slides will progress automatically at the given interval
+    // - false:  No auto-sliding, even if data-autoslide is present
+    autoSlide?: 0 | 1 | false, // 1:
+    // Use this method for navigation when auto-sliding (defaults to navigateNext)
+    /* autoSlideMethod: null, // ToDo: I think this is not needed */
+    // Stop auto-sliding after user input
+    autoSlideStoppable?: boolean,
+    // Transition style for full page slide backgrounds
+    backgroundTransition?: Transition,
+    // Vertical centering of slides
+    center?: boolean,
+    // Display presentation control arrows
+    controls?: boolean,
+    // Visibility rule for backwards navigation arrows
+    controlsBackArrows?: ControlsBackArrows,
+    // Determines where controls appear
+    controlsLayout?: ControlsLayout,
+    // Help the user learn the controls by providing hints, for example by
+    // bouncing the down arrow when they first encounter a vertical slide
+    controlsTutorial?: boolean,
+    // Specify the average time in seconds that you think you will spend
+    // presenting each slide. This is used to show a pacing timer in the
+    // speaker view
+    defaultTiming?: number | null,
+    /* disableLayout?: boolean, // ToDo: I think this is not needed */
```

Comment:
> Disables the default reveal.js slide layout (scaling and centering)
> so that you can use custom CSS layout

Do we want to omit CSS completely? I guess with the iframe it would be possible to define ones own theme in HedgeDoc…